cli/cmd/source/drop.go

```diff
@@ -1,23 +1,67 @@
 package source
 
 import (
+	"context"
 	"fmt"
-	"strings"
+	"regexp"
 
+	"github.com/rilldata/rill/cli/pkg/local"
+	"github.com/rilldata/rill/runtime/artifacts/artifactsv0"
 	"github.com/spf13/cobra"
 )
 
 // dropCmd represents the drop command, it requires min 1 args as source path
-func DropCmd() *cobra.Command {
+func DropCmd(ver string) *cobra.Command {
+	var olapDriver string
+	var olapDSN string
+	var projectPath string
+	var verbose bool
+
 	var dropCmd = &cobra.Command{
 		Use:   "drop <source>",
 		Short: "Drop a source",
-		Args:  cobra.MinimumNArgs(1),
+		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Printf("drop called with args:%s", strings.Join(args, " "))
+			sourceName := args[0]
+
+			if ok, err := regexp.Match("^[ _a-zA-Z0-9\\-]+$", []byte(sourceName)); !ok || err != nil {
+				return fmt.Errorf("not a valid source name: %s", sourceName)
+			}
+
+			app, err := local.NewApp(ver, verbose, olapDriver, olapDSN, projectPath)
+			if err != nil {
+				return err
+			}
+
+			if !app.IsProjectInit() {
+				return fmt.Errorf("not a valid Rill project")
+			}
+
+			repo, err := app.Runtime.Repo(context.Background(), app.Instance.ID)
+			if err != nil {
+				panic(err) // Should never happen
+			}
+
+			c := artifactsv0.New(repo, app.Instance.ID)
+			sourcePath, err := c.DeleteSource(context.Background(), sourceName)
+			if err != nil {
+				return err
```

Comment:
Would be good to beautify the errors a bit. It will saw file doesnt exist instead of source doesnt exist currently. Perhaps it is a bigger issue with reconcile itself?

Reply:
Added some context to the returned errors, but not going in depth with parsing specific errors. The backlog has a separate project for pretty errors, I think we should tackle it there (as there are many unpretty errors you can encounter throughout the project)